### PR TITLE
[ROCm] Add ROCm5.3 to python package pipeline

### PR DESCRIPTION
**Pull request by PeixuanZuo**
### Description

1. Remove ROCm5.1.1 and ROCm5.2 from ROCm python package pipeline
2. Add ROCm5.3 to ROCm python package pipeline
pipeline:
https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=237172&view=results

### CODE REVIEW

**Comment by cloudhan** [comment]
@gopitk Could you please endorse this change and justify the agreement we mada in internal teams chat.

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3 [hunk 1]
```diff
@@ -1,12 +1,108 @@
-ARG BASEIMAGE=rocm/pytorch:rocm5.2_centos7_py3.7_pytorch_staging
+ARG BASEIMAGE=centos:7
 ARG POLICY=manylinux2014
 ARG PLATFORM=x86_64
 ARG DEVTOOLSET_ROOTPATH=
 ARG LD_LIBRARY_PATH_ARG=
 ARG PREPEND_PATH=
 
+# rocm/dev-centos-7:5.3 is what we want to use as base image, but ROCm team hasn't provided it yet.
```
Comment (ytaous) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3:8]: Do u know when will it be provided? It would nice to use it directly.

Comment (PeixuanZuo) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3:8]: I check again, they provided it two days ago. I will use it directly.

**Comment by gopitk** [comment]
Yes. I approve this change to stop the pipeline for ROCm 5.1.1 and 5.2 and have only ROCm 5.2.3 and 5.3 till next release.

**Review by ytaous — APPROVED** [review]

**Comment by ytaous** [comment]
NIT: should we also move 5.1.1 from stable to deprecated stable?
https://download.onnxruntime.ai/

**Comment by baijumeswani** [comment]
Is this change needed for ort 1.13 release?

**Review by baijumeswani — DISMISSED** [review]

**Comment by ytaous** [comment]
> NIT: should we also move 5.1.1 from stable to deprecated stable? https://download.onnxruntime.ai/

Talked to Baiju, let's leave it for now.

**Comment by PeixuanZuo** [comment]
> > Is this change needed for ort 1.13 release?
> 
> I am not sure about this. @ytaous do you know how to confirm this?

Talked to Ethan, this change isn't needed for ort 1.13 release now.

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3 [hunk 2]
```diff
@@ -18,7 +18,17 @@ LABEL maintainer="The ManyLinux project"
...
 
-RUN yum remove -y devtoolset\* git\* && conda remove -y cmake
+# CMake
+ENV CMAKE_VERSION=3.24.2
+RUN cd /usr/local && \
+    wget -q -O - https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz | tar zxf -
+ENV PATH=/usr/local/cmake-${CMAKE_VERSION}-linux-x86_64/bin:${PATH}
+
+# rocm lib
+RUN yum install -y libpython3.8-dev python3-pip miopen-hip-devel rocblas-devel && \
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3:28]: Why do you install libpython3.8-dev? Manylinux docker images should not have this.

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3:28]: And where does this package from? Does Centos ever have this package?

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3:28]: And why do you install "python3-pip"?

Comment (PeixuanZuo) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_rocm5.3]: Remove them.

**Review by ytaous — APPROVED** [review]

**Pull request merged**